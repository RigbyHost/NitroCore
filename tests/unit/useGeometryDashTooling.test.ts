describe("useGeometryDashTooling()", () => {
    const t = useGeometryDashTooling()

    it("Clears GD requests correctly", () => {
        expect(t.clearGDRequest("Mach:ine")).toBe("Mach")
        expect(t.clearGDRequest("Mach|ine")).toBe("Mach")
        expect(t.clearGDRequest("Mach~ine")).toBe("Mach")
        expect(t.clearGDRequest("Mach#ine")).toBe("Mach")
        expect(t.clearGDRequest("(Mac\0h)ine")).toBe("(Mach")
    })

    it("XORs correctly", () => {
        expect(t.doXOR("mogging", "1234567")).toBe("\\]TS\\XP")
    })

    it ("Transforms date correctly", () => {
        const date = new Date()
        expect(t.getDateAgo(date.setSeconds(date.getSeconds() - 1))).toBe("1 seconds")
        expect(t.getDateAgo(date.setMinutes(date.getMinutes() - 1))).toBe("1 minutes")
        expect(t.getDateAgo(date.setHours(date.getHours() - 1))).toBe("1 hours")
        expect(t.getDateAgo(date.setDate(date.getDate() - 1))).toBe("1 days")
        expect(t.getDateAgo(date.setDate(date.getDate() - 10))).toBe("1 weeks")
        expect(t.getDateAgo(date.setMonth(date.getMonth() - 1))).toBe("1 months")
        expect(t.getDateAgo(date.setFullYear(date.getFullYear() - 1))).toBe("1 years")
    })

    it ("Hashes solo correctly", () => {
        // Bruv this shit CANNOT be tested properly, doing this for coverage
        expect(t.hashSolo("Mach:ine")).toMatchSnapshot()
        expect(t.hashSolo2("Mach:ine")).toMatchSnapshot()
        expect(t.hashSolo3("Mach:ine")).toMatchSnapshot()
        expect(t.hashSolo4("Mach:ine")).toMatchSnapshot()
    })

    it("Does GJP correctly", () => {
        // Idk at least it's fucking consistent
        expect(t.doGJP("Mach:ine")).toMatchSnapshot()
        expect(t.doGJP2("Mach:ine")).toMatchSnapshot()
    })

    it ("Parses GD Version correctly", async () => {
        const form = new FormData()
        expect(await t.getGDVersionFromBody(form)).toBe(21)
        form.set("gameVersion", "10")
        expect(await t.getGDVersionFromBody(form)).toBe(10)
        form.set("gameVersion", "20")
        form.set("binaryVersion", "27")
        expect(await t.getGDVersionFromBody(form)).toBe(20)
        form.set("binaryVersion", "28")
        expect(await t.getGDVersionFromBody(form)).toBe(21)
        form.set("gameVersion", "21")
        form.set("binaryVersion", "37")
        expect(await t.getGDVersionFromBody(form)).toBe(22)
    })
})