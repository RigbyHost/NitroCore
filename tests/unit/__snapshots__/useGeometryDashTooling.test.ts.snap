// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGeometryDashTooling() > Does GJP correctly 1`] = `"¸A"`;

exports[`useGeometryDashTooling() > Does GJP correctly 2`] = `"5d7e97a0ba243e789af754a88048d532023f47df"`;

exports[`useGeometryDashTooling() > Hashes solo correctly 1`] = `"7201c5c18f5ad9ebfb4ad80a1b9e9ec54a21bb3b"`;

exports[`useGeometryDashTooling() > Hashes solo correctly 2`] = `"4316d5399e5610b6f15c41b07f454754673b5a81"`;

exports[`useGeometryDashTooling() > Hashes solo correctly 3`] = `"cf014d91f693410c0c473be01fc969a145990b64"`;

exports[`useGeometryDashTooling() > Hashes solo correctly 4`] = `"2b93669e7fe76b8814c3372340fa4c0e1ea4c584"`;
